test-data/unit/check-dynamic-typing.test

```diff
@@ -201,10 +252,12 @@ from typing import Any
 d = None # type: Any
 a = None # type: A
 
-a = d[a()] # E: "A" not callable
+if int():
+    a = d[a()] # E: "A" not callable
 d[a()] = a # E: "A" not callable
 
-a = d[a]
+if int():
+    a = d[a]
```

Comment:
I am also not sure why it is needed here. There is no error on this line. Will this create a new variable with type `Any` that shows the one with type `A`?